stores/cloudflare/src/storage/index.ts

```diff
@@ -0,0 +1,440 @@
+import { type StorageThreadType, type MessageType } from '@mastra/core/memory';
+import {
+  MastraStorage,
+  type TABLE_NAMES,
+  type StorageColumn,
+  type StorageGetMessagesArg,
+  type EvalRow,
+} from '@mastra/core/storage';
+import { TABLE_THREADS } from '@mastra/core/storage';
+import { type WorkflowRunState } from '@mastra/core/workflows';
+
+export interface CloudflareConfig {
+  accountId: string;
+  namespaceId: string;
+  apiToken: string;
+}
+
+export class CloudflareStore extends MastraStorage {
+  private accountId: string;
+  private namespaceId: string;
+  private apiToken: string;
+  private baseUrl: string;
+
+  constructor(config: CloudflareConfig) {
+    super({ name: 'Cloudflare' });
+    this.accountId = config.accountId;
+    this.namespaceId = config.namespaceId;
+    this.apiToken = config.apiToken;
+    this.baseUrl = `https://api.cloudflare.com/client/v4/accounts/${this.accountId}/storage/kv/namespaces/${this.namespaceId}`;
+  }
```

Comment:
We need to be able to separate different types of data into different namespaces.

E.g. Evals, Traces, etc

I believe it would be best to allow users to pass in a namespace prefix, and we create the namespaces for them. Please take a look at [PG Store](https://github.com/mastra-ai/mastra/blob/main/stores/pg/src/storage/index.ts#L171)

The idea is that if we don't separate each type of data into a separate namespace, we have an issue where different data might get mixed together.

For example you can use the cloudflare client sdk to create namespaces with prefixes, and insert values into that namespace

```
     this.client.kv.namespaces.values.update(`mastra_evals`, `k`, {
      metadata: '',
      value: 'v',
      account_id: 'x',
    });
  ```